### PR TITLE
fix: Return tuple array from describe

### PR DESCRIPTION
**Pull request by willium**

### CODE REVIEW

File: src/addon/result-iterator.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class ResultIterator<T> implements IterableIterator<T> {
   /**
    * Describe the result set schema.
    */
-  public describe(): string[][] {
+  public describe(): [string, string][] {
```
Comment (rostislavdeepcrawl) [inline src/addon/result-iterator.ts:36]: if you could change this to `Array<[string, string]>` eslint will be happy :)

Comment (willium) [reply src/addon/result-iterator.ts]: done

File: src/addon/result-iterator.ts [hunk 2]
```diff
@@ -27,13 +27,13 @@ export class ResultIterator<T> implements IterableIterator<T> {
    * @remarks
    * Note, this may produce a `heap out of bounds` error in case when there is too much data. Either use the {@link ResultIterator.fetchRow | fetchRow} or the  {@link Connection.execute | Connection.execute} method when there is a lot of data.
    */
-  public fetchAllRows(): T[] {
+  public fetchAllRows(): readonly T[] {
```
Comment (rostislavdeepcrawl) [inline src/addon/result-iterator.ts:30]: don't think it makes sense to make this readonly (and same for `describe`)

File: src/addon/result-iterator.ts [hunk 3]
```diff
@@ -27,13 +27,13 @@ export class ResultIterator<T> implements IterableIterator<T> {
...
     return <T[]>[...this];
   }
   /**
    * Describe the result set schema.
    */
-  public describe(): string[][] {
+  public describe(): Readonly<Array<[string, string]>> {
```
Comment (rostislavdeepcrawl) [inline src/addon/result-iterator.ts:36]: The `ResultIteratorClass`'s `describe` should also be changed

**Comment by dforsber** [comment]
@willium since I can't edit this PR directly, I opened a new one that implements these changes :).
https://github.com/deepcrawl/node-duckdb/pull/125

**Pull request closed without merging**